test/test_sched.rs

```diff
@@ -30,3 +38,82 @@ fn test_sched_affinity() {
     // Finally, reset the initial CPU set
     sched_setaffinity(Pid::from_raw(0), &initial_affinity).unwrap();
 }
+
+#[test]
+#[cfg(not(any(target_env = "musl", target_os = "freebsd")))]
+fn test_sched_scheduler_not_realtime() {
+    let initial_scheduler = sched_getscheduler(None).unwrap();
```

Comment:
These two tests could interfere with each other.  To prevent that, you should create a SCHED_MTX in test/test.rs and grab it during these tests.

Reply:
Good catch; they can also interfere with the other two new tests if they change the scheduler + params between reads. I've added the mutex to all four.